environments/personalization.md

```diff
@@ -40,35 +42,67 @@ sudo apt-get install -y fish
 sudo chsh -s /usr/bin/fish $USER
 ```
 
-The Environments page shows the log output of the `~/personalize` script in
-the build log.
+The following is a more extensive example of a `~/personalize` script:
+
+```bash
+#!/bin/bash
+
+###########################################################################
+# For use with an environment build using an image that includes git. This
+# script configures git using Coder's personalize script. This script runs
+# each time the environment is rebuilt. The script must be located at
+# ~/personalize. The initial environment will not contain this script, so
+# it must be added after creation.
+###########################################################################
+
+# Backup existing gitconfig if it exists
+if [ -f ~/.gitconfig ]; then
+echo "Backing up ~/.gitconfig"
+mv ~/.gitconfig ~/.gitconfig.bak
+fi
+
+# Set name and email in git
+echo "[user]\n\temail = youremailhere@gmail.com\n\tname = Your Name" > ~/.gitconfig
+```
```

Comment:
I wonder if stuff like this should be eventually be hosted in github gists or in a source-controlled "examples" repo and somehow embedded into the doc? Then, it could maybe be "expanded" too for cleanliness.

Not saying, we should do this now, but would love to hear your thoughts.

Reply:
Definitely think that's something to consider in the future; I think have examples of how we do things would be helpful for our users, especially when it comes to gotchas and the like -- internal tricks and stuff we've learned.